File: backends/go-gin/internal/domain/staff/value_objects.go
package staff

import "time"

// ListItem はスタッフ一覧レスポンス用の値オブジェクトです。
type ListItem struct {
	ID        uint
	Name      string
	Email     string
	Role      int
	Status    int // 0=削除済み, 1=有効
	CreatedAt time.Time
	UpdatedAt time.Time
}

// Vo はスタッフ詳細レスポンス用の値オブジェクトです。
type Vo struct {
	ID     uint
	Name   string
	Avatar *string
	Role   int
}

File: backends/kotlin-ktor/src/main/kotlin/com/authorization/domain/staff/ValueObjects.kt
/*
 * スタッフドメインの値オブジェクトモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.domain.staff

import java.time.LocalDateTime

/**
 * スタッフ一覧行の値オブジェクトです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class ListItem(
    val id:        Long,
    val name:      String,
    val email:     String,
    val role:      Int,
    val status:    Int,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
)

/**
 * スタッフ詳細の値オブジェクトです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class Vo(
    val id:     Long,
    val name:   String,
    val avatar: String?,
    val role:   Int,
)
